src/codegen/ast-substitutions.lisp

```diff
@@ -38,9 +38,31 @@
 
   (:method (subs (node node-lisp))
     (declare (type ast-substitution-list subs)
-             (ignore subs)
              (values node))
-    node)
+    (multiple-value-bind (let-bindings lisp-var-bindings)
```

Comment:
The bindings generated here are very similar to the inlining case
https://github.com/coalton-lang/coalton/pull/1029/files#diff-bfb68d1f5b43842fcc3cf2f8161617a2e306044bcb45ae3a30d6c2a030175857R341